src/lib/comms/include/sol-netctl.h

```diff
@@ -553,6 +696,81 @@ sol_netctl_find_service_by_name(const char *service_name)
 }
 
 /**
+ * @brief register a agent for network connection
+ *
+ * Register the agent for network connection.
+ * The register agent will inform the all information of network connection
+ * via the register callback struct.
+ * It includes the login input, the network connection failure and so on.
+ * The callback struct must be registered.
+ *
+ * @see struct sol_netctl_agent
+ *
+ * @param agent the agent Callback struct to be called when the related information is updated.
```

Comment:
you should say a single agent is registered for an application, that registering a new agent unregisters the previous and to unregister the last one you should use `NULL`.